library/std/src/sys/fs/vexos.rs

```diff
@@ -0,0 +1,600 @@
+use crate::ffi::OsString;
+use crate::fmt;
+use crate::fs::TryLockError;
+use crate::hash::Hash;
+use crate::io::{self, BorrowedCursor, IoSlice, IoSliceMut, SeekFrom};
+use crate::path::{Path, PathBuf};
+use crate::sys::common::small_c_string::run_path_with_cstr;
+use crate::sys::time::SystemTime;
+use crate::sys::{unsupported, unsupported_err};
+
+#[expect(dead_code)]
+#[path = "unsupported.rs"]
+mod unsupported_fs;
+pub use unsupported_fs::{
+    DirBuilder, FileTimes, canonicalize, link, readlink, remove_dir_all, rename, rmdir, symlink,
+    unlink,
+};
+
+#[derive(Debug)]
+struct FileDesc(*mut vex_sdk::FIL);
+
+pub struct File {
```

Comment:
`File` isn't `Send`/`Sync` on this target, but it is on [other targets](https://doc.rust-lang.org/stable/std/fs/struct.File.html#impl-Send-for-File). There might be other places where this is the case, but I haven't run into them.